### PR TITLE
Add Bfdopen() to libbio

### PR DESCRIPTION
**Pull request by gmacd**
Required for git9.
From plan9port via 9front (commit 035aacf6f5c23f0513e240046818ef203e8d87cb).

Signed-off-by: Graham MacDonald <grahamamacdonald@gmail.com>

### CODE REVIEW

File: sys/man/2/bio [hunk 1]
```diff
@@ -93,6 +96,17 @@ It calls
...
+.I Bfdopen
+allocates a buffer for the already-open file descriptor
+.I fd
+for mode
+.B OREAD
+or
+.BR OWRITE .
+It calls
+.IR malloc (2)
+to allocate a buffer.
```
Comment (dancrossnyc) [inline sys/man/2/bio:108]: s/a buffer/the buffer/

File: sys/src/libbio/binit.c [hunk 2]
```diff
@@ -86,35 +85,55 @@ Binits(Biobufhdr *bp, int f, int mode, uchar *p, int size)
...
-	Binits(bp, f, mode, bp->b, sizeof(bp->b));
-	bp->flag = Bmagic;			/* mark bp open & malloced */
+	if(fd < 0)
+		return nil;
+	bp = Bfdopen(fd, mode);
+	if(bp == nil) {
+		close(fd);
+		return nil;
+	}
+	setmalloctag(bp, getcallerpc(&name));
```
Comment (dancrossnyc) [inline sys/src/libbio/binit.c:135]: Why do `setmalloctag` here? That is, I understand why one would do so, but perhaps add a comment explaining the rationale?

**Review by dancrossnyc — APPROVED** [review]
A couple of minor things; otherwise, LGTM.

**Comment by gmacd** [comment]
Let me know your thoughts @dancrossnyc

**Review by dancrossnyc — APPROVED** [review]

**Pull request merged**